### DISCUSSION TITLE
A possible inconsistency in the Typst writer

### DISCUSSION THREAD

**Discussion by bcdavasconcelos** [jgm/pandoc#9309] [Q&A] [answered]
First of all, I am happy to see Pandoc Discuss reemerge after its demise in Google Discussions. 

I *probably* noticed an inconsistency in the Typst writer, which does not occur with citeproc `true`, but that occurs when using an alternative to citeproc which **should** work in the same way. 

## Lua filter

```lua
function Pandoc (doc)
  return pandoc.utils.citeproc(doc)
end
```

## Source document

```markdown
## Heading

Text [@AristOp].

---
references:
- author:
  - family: Aristotle
  edition: 1
  editor:
  - family: Bekker
    given: Immanuel
  id: AristOp
  issued: 1831
  number-of-volumes: 4
  publisher: Reimer
  publisher-place: Berlim
  title: Aristotelis opera
  type: book
---
```

## to HTML

```html
<section id="heading" class="level2">
<h2 class="anchored" data-anchor-id="heading">Heading</h2>
<p>This filter adds formatting to heading text <span class="citation" data-cites="AristOp">(<a href="#ref-AristOp" role="doc-biblioref">Aristotle 1831</a>)</span>.</p>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-AristOp" class="csl-entry" role="listitem">
Aristotle. 1831. <em>Aristotelis Opera</em>. Edited by Immanuel Bekker. 1st ed. 4 vols. Berlim: Reimer.
</div>
</div>
</section>
```

## to Typst

```typst
== Heading
<heading>
This filter adds formatting to heading text #cite(<AristOp>);.

#block[
#block[
Aristotle. 1831. #emph[Aristotelis Opera];. Edited by Immanuel Bekker. 1st ed. 4 vols. Berlim: Reimer.

] <ref-AristOp>
] <refs>
```

Notice how the inline citation gets rendered in the HTML, but **does not** in the Typst output. I can't think of any reason for this being the case.

Am I overlooking something here?

**Comment by jgm** [comment, marked as answer]
If the `citations` extension is enabled (default for typst), typst will emit `#cite(..)`; otherwise, it will emit the formatted citations.  So, try `-t typst-citations`.

**Reply by bcdavasconcelos** [reply]
Works perfectly, @jgm. Thank you.

**Comment by iandol** [comment]
@bcdavasconcelos I made a pull request to have this documented: #9127 — I wasn't sure the best place for this information...

**Reply by bcdavasconcelos** [reply]
That is a good idea, @iandol. Personally, I spent hours toiling away with several Lua filters that never had any chance of solving the issue. Should it not be that whatever needs to happen to get citeproc working does when the command mentioned runs in a filter? I would think so, but this is arguably a very fringe case where the user has `citeproc: false` but expects citeproc to end up running; moreover, I have a pretty incomplete picture of the whole Pandoc library, so there might be a reason for this way of working, which I could not fully understand. With all of that in mind, it is at least a good idea to make it more explicit and documented

**Reply by jgm** [reply]
It behaves the same whether you use the filter or command line `--citeproc`.  The behavior is exactly the same as the longstanding behavior of `markdown`: `-t markdown+citations` will use `@foo`, `-t markdown-citations` will resolve the citation.  Similarly for `org` and `docx`.

**Reply by bcdavasconcelos** [reply]
Thank you for the clarification, @jgm. It makes a whole lot more sense now that I saw both options together (`+citations`, `-citations`). I had forgotten extensions are added using a `+` sign instead of simply joining them using `-`, hence my original estrangement.
